### ISSUE TITLE
Animated stickers without using third party services

### ISSUE THREAD

**Issue by hbarcellos** [open-wa/wa-automate-nodejs#868]
Hi,

It would be great to find out a way to send animated stickers without using third-party services (like giphy).

To make sure people don't make any mistake sending the stickers incorrectly converted we can provide guidelines for conversion.

**Comment by rebahanelite** [comment]
You can use api upload to giphy, and combine it with sendGiphyAsSticker, it works on my bot

**Comment by hbarcellos** [comment]
Yes, I already do that too, but it is not good to depend on third party services.
You need to store the stickers on your giphy account, you can't create any stickers that violate their policies (NSFW), all of your uploads should be public, you have API rate limits...

It would be great to have conversion to correct format without using giphy.

**Comment by smashah** [comment]
Wow in one issue someone wants to mass report nsfw accounts and in another issue a feature is being requested to create nsfw animated stickers. Truly a dilemma.

**Comment by hbarcellos** [comment]
Whatsapp allows NSFW.

**Comment by hbarcellos** [comment]
Perfect! A self-hosted solution solves the problem.

**Comment by smashah** [comment]
Just FYI, this feature is ready but I wanted to make it so the gifs are properly by cropping, padding and centering. This is not possible right now without a consulted custom libvip build so I'm just going to implement it as is warts and all.

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i`);
exec(`npm run release-ci`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [2.0.27](https://github.com/open-wa/wa-automate-nodejs/compare/2.0.26...2.0.27) (2020-10-02)

- build(deps): bump update-notifier from 4.1.1 to 5.0.0 [`#877`](https://github.com/open-wa/wa-automate-nodejs/pull/877)
- build(deps): bump open from 7.2.1 to 7.3.0 [`#878`](https://github.com/open-wa/wa-automate-nodejs/pull/878)
- build(deps): bump hasha from 5.2.0 to 5.2.1 [`#871`](https://github.com/open-wa/wa-automate-nodejs/pull/871)
- build(deps-dev): bump marked from 1.1.1 to 1.2.0 [`#872`](https://github.com/open-wa/wa-automate-nodejs/pull/872)
- Release 2.0.27 [`279fff0`](https://github.com/open-wa/wa-automate-nodejs/commit/279fff0e78be8dd877a71381757a91364d3e537b)
- Added demo for sending file from relative file path #876 [`87d8c46`](https://github.com/open-wa/wa-automate-nodejs/commit/87d8c46ede3c2b6b35a71244ecfc78e6198f2e83)
- Feature: Implemented automatic animated sticker conversion using existing methods #868 [`25e8870`](https://github.com/open-wa/wa-automate-nodejs/commit/25e8870612e9834136f739db02b685bf92781444)
- Implemented `client.reportSpam` [restricted] #867 [`0966717`](https://github.com/open-wa/wa-automate-nodejs/commit/096671713b8b01c74e4914937620076739d7eddd)
- Checks file regex if valid DataURL + better error handling [`afd31c5`](https://github.com/open-wa/wa-automate-nodejs/commit/afd31c552e049246f65d4fe65f9e710395035763)
- Demo improvements [`43b9aa0`](https://github.com/open-wa/wa-automate-nodejs/commit/43b9aa0ef87da13ce0e309d14b5f0f6553585bcb)
- Patches Update [`50b4efc`](https://github.com/open-wa/wa-automate-nodejs/commit/50b4efc705a7a54a0f6394a085b0a8fe871da2ad)
- Added note in docs to make sure a single normal message is sent before trying to send a reply #869 [`af2633a`](https://github.com/open-wa/wa-automate-nodejs/commit/af2633a9501f0a0f1fea0d342e8c34921df46c91)
- fix typo in demo [`9f31f93`](https://github.com/open-wa/wa-automate-nodejs/commit/9f31f93fe4f2524ac04407e68be1460c963eabed)

**Comment by smashah** [comment]
@hbarcellos @rebahanelite 

Please use the latest version and use the existing sticker methods to send animated gifs. Animated gifs are detected automatically so no need to change your code to add a new param. 

https://github.com/open-wa/wa-automate-nodejs/blob/279fff0e78be8dd877a71381757a91364d3e537b/demo/index.ts#L165

 Happy stickering!
